functional/transformations.py

```diff
@@ -143,7 +143,8 @@ def distinct_t():
     """
 
     def distinct(sequence):
-        return iter(set(sequence))
+        seen = set()
+        return iter([x for x in sequence if not (x in seen or seen.add(x))])
```

Comment:
nit: I think I'd prefer the multiline version if the conditional has side effects (.add()).

Another possibility is to use this opportunity to make distinct() lazy by using `yield` like here https://github.com/EntilZha/PyFunctional/blob/master/functional/transformations.py#L253

I think it would be as simple as something like
```
seen = set()
for x in sequence:
    if x in seen:
        continue
    else:
        seen.add(x)
        yield x

Reply:
Solid suggestion. 

Ya I could do that. I had a version like that earlier, but it wasn't lazy.

I like that idea a lot. I will move to something like that.

Reply:
There we are. I pushed a change similar to that.